controllers/platform/services/services.go

```diff
@@ -198,11 +198,24 @@ func (d DataIndexHandler) MergePodSpec(podSpec corev1.PodSpec) (corev1.PodSpec,
 	return *c, err
 }
 
+// hasPostgreSQLConfigured returns true when either the SonataFlow Platform PostgreSQL CR's structure or the one in the Data Index service specification is not nil
+func (d DataIndexHandler) hasPostgreSQLConfigured() bool {
+	return (d.platform.Spec.Services.DataIndex.Persistence != nil && d.platform.Spec.Services.DataIndex.Persistence.PostgreSQL != nil) ||
+		(d.platform.Spec.Persistence != nil && d.platform.Spec.Persistence.PostgreSQL != nil)
+}
```

Comment:
i wonder if this should be enhanced by adding `d.IsServiceSetInSpec` to the check as well. like this?
```golang
	return (d.IsServiceSetInSpec && d.platform.Spec.Services.DataIndex.Persistence != nil && d.platform.Spec.Services.DataIndex.Persistence.PostgreSQL != nil) ||
		(d.platform.Spec.Persistence != nil && d.platform.Spec.Persistence.PostgreSQL != nil)
```

Reply:
my suggestion is more about avoiding a null pointer exception. how can you check if `d.platform.Spec.Services.DataIndex.Persistence != nil` when `d.platform.Spec.Services.DataIndex == nil`?

for the same reason you're already checking `d.platform.Spec.Services.DataIndex.Persistence` before you check `d.platform.Spec.Services.DataIndex.Persistence. PostgreSQL`

Reply:
> at this point of the execution the DI service has been already established that it exists, otherwise the parent function would not be called in the first place.

I'm not sure that's accurate though. Maybe at the point you're currently using `hasPostgreSQLConfigured` ... but we can't assume that will always be the case. This function should work regardless of where/when it's called imo